### PR TITLE
Fix/check execute decimals

### PR DESCRIPTION
**Pull request by aelmanaa**
This pull request introduces significant improvements to how token amount denominations are handled across chains, especially when source and destination tokens have differing decimal places. The changes ensure that token amounts are correctly converted and validated according to the token's decimals, preventing silent errors and improving error reporting. Additionally, the codebase is refactored for better test coverage and utility reuse.

**Token denomination handling and validation:**

* Updated `Chain.checkExecute` to convert source-denominated token amounts to destination token decimals before comparing against pool balances and rate limits. This prevents transfers that would result in unrepresentable or zero-value releases on the destination chain. A new error, `CCIPTokenDecimalsInsufficientError`, is thrown if the conversion results in an unrepresentable amount. [[1]](diffhunk://#diff-807a4ca9474665a0a92dd38d0c7b802fa323550a99ef0c98a2ae754be2cd8838L1393-R1462) [[2]](diffhunk://#diff-807a4ca9474665a0a92dd38d0c7b802fa323550a99ef0c98a2ae754be2cd8838R17)
* Refactored the logic for extracting source decimals from `extraData` and for scaling token amounts into utility functions (`getSourceDecimalsFromExtraData`, `scaleDecimals`), and reused these utilities throughout the codebase for consistency. [[1]](diffhunk://#diff-807a4ca9474665a0a92dd38d0c7b802fa323550a99ef0c98a2ae754be2cd8838L66-R74) [[2]](diffhunk://#diff-dc947dd345124b95450aec2428d6a45de93bdeebcfeaa47014612805a8191098L20-R20) [[3]](diffhunk://#diff-dc947dd345124b95450aec2428d6a45de93bdeebcfeaa47014612805a8191098L177-L188) [[4]](diffhunk://#diff-dc947dd345124b95450aec2428d6a45de93bdeebcfeaa47014612805a8191098L234-R222)

**Testing improvements:**

* Added a comprehensive new test suite in `check-execute.test.ts` to verify correct denomination handling, conversion, error reporting, and edge cases (such as zero or mismatched decimals).
* Extended EVM chain tests to verify that denomination conversions are only applied for generic checks and not for on-chain simulations, and clarified test comments for rate-limit behavior. [[1]](diffhunk://#diff-efabde28ff377c08cdfad8f0a886bfab6075757a2f37352edda8df6e5d6e0a54R893-R918) [[2]](diffhunk://#diff-efabde28ff377c08cdfad8f0a886bfab6075757a2f37352edda8df6e5d6e0a54L1001-R1029)

**Error handling and documentation:**

* Improved error context and messaging for rate limit and insufficient balance errors to include both source and destination units, as well as decimal information, aiding in debugging and integration.
* Updated comments and error handling in the EVM chain implementation to clarify when generic checks defer to on-chain simulation for verdicts, and to document the limitations of generic checks for certain pool types and versions.

**Utility and import refactoring:**

* Cleaned up imports and removed redundant local implementations of utility functions, consolidating logic in shared utilities. [[1]](diffhunk://#diff-807a4ca9474665a0a92dd38d0c7b802fa323550a99ef0c98a2ae754be2cd8838L1-R1) [[2]](diffhunk://#diff-807a4ca9474665a0a92dd38d0c7b802fa323550a99ef0c98a2ae754be2cd8838L34-R35) [[3]](diffhunk://#diff-dc947dd345124b95450aec2428d6a45de93bdeebcfeaa47014612805a8191098L1-R1) [[4]](diffhunk://#diff-a5fe9cf02e24c7530c13e210715967b5478159e2f1d8b5006f4db208acc0662cR19-R26)

These changes collectively improve the correctness, maintainability, and reliability of cross-chain token transfers in the SDK.

### CODE REVIEW

File: ccip-sdk/src/chain.ts [hunk 1]
```diff
@@ -1390,36 +1398,68 @@ export abstract class Chain<F extends ChainFamily = ChainFamily> {
...
-        ])
-        if (balance < amount) {
-          throw new CCIPInsufficientBalanceError(balance.toString(), amount.toString(), symbol, {
-            context: { tokenPool, token, typeAndVersion, network: this.network.name },
-          })
+
+      // `ta.amount` is source-denominated whenever `extraData` declares the source decimals.
+      // Convert for the comparisons below, but never touch `ta.amount`: EVMChain's override
+      // forwards it to releaseOrMint as sourceDenominatedAmount.
+      const isLockRelease = Boolean(typeAndVersion?.includes('LockRelease'))
```
Comment (andrevmatos) [inline ccip-sdk/src/chain.ts]: Pass this `typeAndVersion` (if present) through `parseTypeAndVersion` and use the first returned element (`type`); because the string returned by getTokenpoolConfig is the raw `typeAndVersion`, and make be on kebab-case in other chains (e.g. Solana, so like `foo-lock-release 1.5.0`), and `type` is already CamelCase-normalized; then, you compare it `.includes('LockRelease')`

File: ccip-sdk/src/utils.ts [hunk 2]
```diff
@@ -286,6 +286,57 @@ export function getDataBytes(data: BytesLike | readonly number[]): Uint8Array {
...
+  return (amount * BigInt(10) ** BigInt(toDecimals)) / BigInt(10) ** BigInt(fromDecimals)
+}
+
+/**
+ * Whether a `typeAndVersion` string's version is below `minVersion`.
+ * @param typeAndVersion - E.g. `'LockReleaseTokenPool 1.6.0'`; an unparseable one reads as not-below.
+ * @param minVersion - Exclusive lower bound, e.g. `'1.6.1'`.
+ * @returns True only when the parsed version is strictly below `minVersion`.
+ */
+export function isVersionBelow(typeAndVersion: string | undefined, minVersion: string): boolean {
```
Comment (andrevmatos) [inline ccip-sdk/src/utils.ts]: I don't think you need this utility; we've been comparing strings on `version` directly, and it should work, i.e. `1.6.0 < 1.6.1`; we don't ever had double digits for version components

**Review by andrevmatos — APPROVED** [review]
